Feature/fulfill the contact information

PSebou opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | First step of new OPC for v9. It's to make the fields firstname, lastname and gender optional
| Type?             | new feature
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Add production on cart and go on /one-page-checkout
| UI Tests          | 
| Fixed issue or discussion?     |  https://github.com/PrestaShop/PrestaShop/discussions/39047
| Related PRs       | theme : https://github.com/PrestaShop/classic-theme/pull/173
| Sponsor company   | no

Hlavtox committed:
Unify error handling

PSebou committed:
modify firstname and lastname on optional fields

Hlavtox committed:
Use proper default type

Hlavtox committed:
Adapt tests

boherm committed:
Fix customer birthdate validation in FO

PSebou committed:
fix php stan

PSebou committed:
fix php stan

M0rgan01 committed:
mergeShipmentModal

boherm committed:
Bump some modules

boherm committed:
Add new contributors and changelog for next 8.2.2

Hlavtox committed:
Comment carrier logic

jolelievre committed:
Check defined service in QueryListProvider and trhow error when it is not the expected interface

jolelievre committed:
Update attribute group default value to be an int not a string

prestonBot committed:
Updated PrestaShop Packages

jolelievre committed:
Merge pull request #39175 from jolelievre/improve-api-errors

Improve API misconfiguration error

jolelievre committed:
Bump patch version 9.0.1

Hlavtox committed:
Improve warning on lang page

Progi1984 committed:
Merge pull request #39150 from PrestaShop/cron-php-update-modules-9.0.x

jolelievre committed:
Merge pull request #39180 from jolelievre/bump-901

Bump patch version 9.0.1

boherm committed:
Merge branch '9.0.x' into 'develop'

# Conflicts:
#	install-dev/install_version.php
#	src/Core/Version.php
#	tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/07_searchAddRemoveVoucher.ts
#	tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/10_checkSummary.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/01_information/01_withAndWithoutCode.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/01_information/02_enableDisablePartialUse.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/01_information/03_priority.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/02_conditions/04_checkTotalAvailable.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/02_conditions/05_checkTotalAvailableForEachUser.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/02_conditions/06_countrySelection.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/02_conditions/07_carrierSelection.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/02_conditions/10_productSelection.ts
#	tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/03_actions/01_applyToSpecificProduct.ts
#	tests/UI/campaigns/functional/FO/classic/10_cart/02_cart/03_addPromoCode.ts
#	tests/UI/campaigns/functional/FO/classic/10_cart/02_cart/05_displayDiscount.ts
#	tests/UI/campaigns/functional/FO/classic/11_checkout/02_displayOfTotals.ts
#	tests/UI/campaigns/functional/FO/hummingbird/10_cart/02_cart/03_addPromoCode.ts
#	tests/UI/campaigns/functional/FO/hummingbird/10_cart/02_cart/05_displayDiscount.ts
#	tests/UI/campaigns/functional/FO/hummingbird/11_checkout/02_displayOfTotals.ts
#	tests/UI/campaigns/functional/FO/hummingbird/11_checkout/03_displayOfHighlightedCartRules.ts

PSebou committed:
OPC controller

PSebou committed:
OPC controller

jolelievre committed:
Merge pull request #39186 from boherm/merge-9.0.x-into-develop

Merge `9.0.x` into `develop`

Hlavtox committed:
Merge pull request #39006 from Hlavtox/verify-if-categories-exist

Verify if category exists before accessing it

nicosomb committed:
Merge pull request #38042 from Codencode/add-carrier-module-name-to-carrier-list

Added field "Module name" to admin carrier list

Hlavtox commented on controllers/front/OnePageCheckoutController.php:
```diff
@@ -0,0 +1,432 @@
+    {
+        parent::postProcess();
+
+        if (Tools::isSubmit('submitReorder')
```
This must be centralized. :-) This is duplicating the code for reoder.

Hlavtox commented on classes/Customer.php:
```diff
@@ -180,8 +180,8 @@ class CustomerCore extends ObjectModel
             'secure_key' => ['type' => self::TYPE_STRING, 'validate' => 'isMd5', 'copy_post' => false, 'size' => 32],
-            'lastname' => ['type' => self::TYPE_STRING, 'validate' => 'isCustomerName', 'required' => true, 'size' => 255],
-            'firstname' => ['type' => self::TYPE_STRING, 'validate' => 'isCustomerName', 'required' => true, 'size' => 255],
+            'lastname' => ['type' => self::TYPE_STRING, 'validate' => 'isCustomerName', 'size' => 255],
```
This was blocked many times in https://github.com/PrestaShop/PrestaShop/pull/38988, why are you opening yet another PR with this change?

matks replied:
Hello @Hlavtox, @PSebou discussed this topic with @kpodemski and @jolelievre and yes indeed this change is impactful. We thought it could be nice at the beginning however following feedbacks, the plan is to change this to allow empty values instead of "not required". Easier for database structure.

Hlavtox commented on controllers/front/OnePageCheckoutController.php:
```diff
@@ -0,0 +1,432 @@
+            'cartUrl' => '',
+        ];
+
+        if ($this->context->cart->isAllProductsInStock() !== true
```
This should be centralized, it's already in the code like 3 times, every implementation checking something else. It would be good to have one source of truth called `isCartOrderable` that will check absolutely everyting, and call it in `CartController` and `OrderController`.

Hlavtox commented on controllers/front/OnePageCheckoutController.php:
```diff
@@ -0,0 +1,432 @@
+            Tools::getAllValues()
+        );
+
+        $presentedCart = $this->cart_presenter->present($this->context->cart, true);
```
Already a second cart present I see during this review, every cart present kills performance, it should be done once per request.

Hlavtox commented on controllers/front/OnePageCheckoutController.php:
```diff
@@ -0,0 +1,432 @@
+        }
+
+        // Check that products are still orderable, at any point in checkout
+        if ($this->context->cart->isAllProductsInStock() !== true
```
One more oderability check

Codencode committed:
Refactor cart recovery: use updateCustomer for restoring customer session

During cart recovery, replaced manual cookie assignments with Context::updateCustomer(),
which restores the customer session and updates CustomerSession properly.

Left `check_cgv` and `id_cart` assignments intact, as they are not managed by updateCustomer.

cnavarro-prestashop committed:
fix(Auto-format): Disable auto format saving

cnavarro-prestashop committed:
fix(auto-format): Disable auto format

cnavarro-prestashop committed:
fix(auto-format): Disable auto-format

cnavarro-prestashop committed:
Update src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/url_schema_configuration.html.twig

Co-authored-by: Daniel Hlaváček <daniel.hlavacek@hotmail.cz>

cnavarro-prestashop committed:
fix(phpcs): PHPCS Fixer

Hlavtox committed:
Add product errors

jolelievre committed:
When enriching the module data always keep the more up to date version available

nicosomb committed:
Merge pull request #39130 from nicosomb/discount-38675

Add trigger on a brand for discount

tleon committed:
chore(discounts): condition based on categories

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions : Apply a discount to Specific product

Hlavtox committed:
Change 404 admin template

Hlavtox committed:
Use translation

Hlavtox committed:
Merge pull request #39153 from Progi1984/issue36596

BO Theme import : Fixed bad display when filename is too long

Codencode committed:
Added PrestaShop version in bin/console

cnavarro-prestashop committed:
fix(RouteValidator): Review PR

tblivet committed:
style: merge modal

jolelievre committed:
Merge pull request #39193 from jolelievre/modules-hooks-higher-version-priority

When enriching the module data always keep most recent version

Hlavtox committed:
Merge pull request #39192 from Hlavtox/add-product-errors

Add product errors

nicosomb committed:
Merge pull request #39008 from boherm/fix-customer-birthDate

Fix customer birthdate validation in FO

Commit:
BO : Invoices : Harmonize Helper Text

kpodemski committed:
Merge pull request #39168 from cnavarro-prestashop/fix/36392

Fix RouteValidator by checking unknown keywords in schema URL

Progi1984 committed:
Merge pull request #39194 from Progi1984/test10323

Codencode committed:
Refactor AboutCommand to extend Symfony's base command and avoid code duplication

Replaced the custom AboutCommand implementation with a subclass of Symfony's BaseAboutCommand
to prevent code duplication and ensure better maintainability as the upstream command evolves.

Codencode committed:
Override getLongVersion to display PrestaShop version in console output

boherm committed:
Merge pull request #39145 from boherm/bump-to-8.2.2

Hlavtox committed:
Merge pull request #39167 from Hlavtox/clarify

Comment carrier logic

jolelievre committed:
Apply filters mapping on the orderBy query parameter in listing

Codencode committed:
phpstan baseline: skip final class extension error in AboutCommand

Codencode committed:
Fix Select2 initialization for new carrier dropdown in modal

Reinitialize the Select2 component on the new carrier select element,
specifying the modal as the dropdown parent to prevent display and
positioning issues within Bootstrap modals.

Commit:
BO : Fixed HTML in informations & warnings block

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin in /tests/UI

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.37.0 to 8.38.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.38.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-version: 8.38.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump dotenv from 17.2.0 to 17.2.1 in /tests/UI

Bumps [dotenv](https://github.com/motdotla/dotenv) from 17.2.0 to 17.2.1.
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v17.2.0...v17.2.1)

---
updated-dependencies:
- dependency-name: dotenv
  dependency-version: 17.2.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

Codencode committed:
Fix: fallback to AdminDashboard if employee default tab is missing

jolelievre committed:
Merge pull request #39206 from jolelievre/mapp-api-orderby

Apply filters mapping on the orderBy query parameter in API listing

boherm committed:
Fix changelog for next 8.2.2

boherm committed:
Merge pull request #39217 from boherm/fix-date-changelog-822

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions : Apply a discount to Cheapest product

Progi1984 committed:
Merge pull request #39212 from PrestaShop/dependabot/npm_and_yarn/tests/UI/develop/typescript-eslint/eslint-plugin-8.38.0

Codencode committed:
Update labels

jolelievre committed:
Fix the result of paginated response, so that filters and orderBy are correctly based on the input from query and not mapped to match grid naming

Progi1984 committed:
Merge pull request #39213 from PrestaShop/dependabot/npm_and_yarn/tests/UI/develop/dotenv-17.2.1

Progi1984 committed:
Merge pull request #39215 from Progi1984/90x_bumpUITestingLib

Progi1984 committed:
Merge pull request #39224 from Progi1984/test10324

Touxten committed:
Fix some informations

Touxten committed:
Update picture

Touxten committed:
Change version back office alt

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions : Apply a discount to Selected product(s)

jolelievre committed:
Merge pull request #39228 from jolelievre/api-orderby-in-response

Fix the result of Admin API paginated response

jolelievre committed:
Merge branch '9.0.x' into merge-develop-90x-290725

jolelievre committed:
Merge pull request #39235 from jolelievre/merge-develop-90x-290725

Merge 9.0.x to develop 29/07/25

tblivet committed:
chore: lint

Progi1984 committed:
Merge pull request #39234 from Progi1984/test10325

M0rgan01 committed:
Add modal loading

M0rgan01 committed:
Add modal loading

matks commented:
Hi @Hlavtox this PR was a draft used by @PSebou to discuss with @kpodemski and @jolelievre to provide an example of what could be done. Lot of changes are coming, please don't review as long as this is a draft 😉 more changes coming

Also what @PSebou did was to make a copy-paster of actual controller, so some items you mentioned like the cart presenter being used and bad for performance come from current codebase. @PSebou is trying to keep the two as close as possible to make it easier for module developers to adapt to One Page Checkout

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions : Exclude discounted products

Progi1984 committed:
Merge pull request #39242 from Progi1984/test10326

tleon committed:
chore(discounts): condition based on supplier

Codencode committed:
Fix: update employees' language to default when a language is deleted

lukaslau committed:
Adding taxes for shipping cost if needed

kpodemski committed:
Merge pull request #35503 from b44x/improve-gd-test

Improve GD test precision - check JPEG support

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions : Send a free gift

jolelievre committed:
Add missing attribute definition on constraints so they can be used in API resources

jolelievre committed:
Improve DefaultLanguage constraint and validator to be usable in Admi API Resources

Hlavtox committed:
Merge pull request #39216 from Codencode/fix-39081-Call-to-a-member-function-getClassName()-on-null-when-the-Employee-default-tab-does-not-exist-anymore

Fix: Call to a member function getClassName() on null when the Employee default tab does not exist anymore

Hlavtox committed:
Merge pull request #38991 from Hlavtox/fix-default-type

Change default product type to TYPE_STANDARD

Hlavtox committed:
Merge pull request #38822 from ShaiMagal/unSerialize

Another level of protection for Tools::unSerialize (allowed_classes => false) + refacto

Progi1984 committed:
Merge pull request #39256 from Progi1984/test10327

Commit:
BO : Admin API : Fixed selectors for copying secret

Commit:
BO : TinyMCE : Fixed checkbox icon

Commit:
BO : Fixed the memorization of the language used in Admin in the cookie (if option is enabled)

Codencode committed:
Added closure of the 'deleteCategoriesForm' form

Codencode committed:
Make the TinyMCEEditor more flexible and extensible via module

Commit:
Command `prestashop:list:commands-and-queries` : Filter CQRS with (1) or without (0) an endpoint

Commit:
BO : Export SQL from Grid : Format SQL

Hlavtox committed:
Fix last positions

jolelievre committed:
Merge pull request #39257 from jolelievre/constraints-attributes

Add missing attribute tag on constraints

jolelievre committed:
Merge branch '9.0.x' into merge-develop-90x-040825

dependabot[bot] committed:
build(deps): bump mysql2 from 3.14.2 to 3.14.3 in /tests/UI

Bumps [mysql2](https://github.com/sidorares/node-mysql2) from 3.14.2 to 3.14.3.
- [Release notes](https://github.com/sidorares/node-mysql2/releases)
- [Changelog](https://github.com/sidorares/node-mysql2/blob/master/Changelog.md)
- [Commits](https://github.com/sidorares/node-mysql2/compare/v3.14.2...v3.14.3)

---
updated-dependencies:
- dependency-name: mysql2
  dependency-version: 3.14.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

Hlavtox committed:
Merge pull request #39182 from Hlavtox/lang-alert-warning

Improve warning on languages page

Hlavtox committed:
Merge pull request #39191 from Codencode/fix-20977-recoverCart-link-does-not-log-the-user.-Stuck-to-process-checkout-step-1

Fix: recoverCart link does not log the user. Stuck to process checkout step 1

Commit:
BO : Fixed redirection when permission is not allowed

jolelievre committed:
Improve Admin API internals so that CQRSDelete can be used like other write operations and with a CQRSCommand, also add an emptyBody extra property that allows empty body when specified

Progi1984 committed:
Merge pull request #39289 from PrestaShop/dependabot/npm_and_yarn/tests/UI/develop/mysql2-3.14.3

Quetzacoalt91 committed:
Merge pull request #39053 from PoulainMaxime/feat/add-missing-generated-password-method

feat: add logic to generate random password on employee profil

Quetzacoalt91 committed:
Update PHPDoc of ActionsBarButtonInterface

Hlavtox committed:
Merge pull request #39285 from Progi1984/issue29908

BO : Export SQL from Grid : Format SQL

Hlavtox committed:
Merge pull request #39160 from Codencode/fix-39059-v9.0.0-Update-order-statuses-crash-when-language-is-disabled

Fix OrderStateType form to show all languages, including disabled ones

Hlavtox committed:
Merge pull request #39208 from Codencode/fix-38961-Carrier-search-not-working-when-editing-order's-carrier

Fix: Carrier search not working when editing order's carrier

Commit:
BO : Products - Bulk Actions : Fixed error when no permissions

kpodemski committed:
Merge pull request #39288 from jolelievre/merge-develop-90x-040825

Merge develop 9.0.x 04/08/25

kpodemski committed:
Merge pull request #39274 from Codencode/fix-39270-missing-closing-/form-tag-in-src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/delete_categories_modal.html.twig

Added closure of the 'deleteCategoriesForm' form

Progi1984 committed:
Merge pull request #39291 from Progi1984/90x_bumpUITestingLibrary

M0rgan01 committed:
orderShipmentsTabContent

jolelievre committed:
Update module ps_apiresources

jolelievre committed:
Adapt serializer test with fixes on Hook API resource

M0rgan01 committed:
Merge pull request #39137 from M0rgan01/create-merge-form

Create merge shipment form

jolelievre committed:
Update UI tests for API

jolelievre committed:
Merge branch '8.2.x' into merge-90x-82x-050825

kpodemski committed:
Merge pull request #39203 from Progi1984/issue30632

BO : Invoices : Harmonize Helper Text

Hlavtox committed:
Merge pull request #39209 from Progi1984/issue36635

BO : Fixed HTML in informations & warnings block

PSebou committed:
update with guest form WIP

PSebou committed:
update with guest form WIP

jolelievre committed:
Use AsDecorator attribute to decorate service

Hlavtox committed:
Merge pull request #39010 from Matt75/fix/checkout-address-step

fix(checkout-address-step): save address choice on click

jolelievre committed:
Merge pull request #39304 from jolelievre/merge-90x-82x-050825

Merge 8.2.x into 9.0.x 05/08/25

jolelievre committed:
Merge pull request #39295 from jolelievre/api-delete-command-empty-body

Admin API CQRSDelete can use CQRSCommand and empty bodies can be used

jolelievre committed:
Merge branch '9.0.x' into merge-90-develop-050825

jolelievre committed:
Merge pull request #39140 from tleon/Issue-38673-discount-condition-based-on-category

Condition based on categories

jolelievre committed:
Merge pull request #39309 from jolelievre/merge-90-develop-050825

Merge 9.0.x to develop 05/08/25

Hlavtox committed:
Merge pull request #39297 from Quetzacoalt91/update-phpdoc-of-ActionsBarButtonInterface

Update PHPDoc of ActionsBarButtonInterface

Hlavtox committed:
Merge pull request #39260 from lukaslau/fix-shipping-tax-9x

Adding taxes for shipping cost if needed

M0rgan01 committed:
Merge pull request #39302 from M0rgan01/fix-shipment-list

Fix shipments tab

Touxten committed:
Update login_layout.html.twig

Commit:
BO : Orders : Modal Discount reset when it's hidden

Hlavtox committed:
Merge pull request #39312 from Touxten/fix-alt-shop-name

Fix ALT shop name.

jolelievre committed:
Merge pull request #39284 from Progi1984/90x_listCommandsAndQueriesCommand

Command `prestashop:list:commands-and-queries` : Filter CQRS with (1) or without (0) an endpoint

PSebou committed:
update with guest form WIP

PSebou committed:
update with guest form WIP

jolelievre committed:
Update Distribution API Client to include the new wall of fame

jolelievre committed:
Update Distribution API Client to include the new wall of fame

Touxten committed:
Fix some informations in README

Fix some informations in README

kpodemski committed:
Merge pull request #39292 from Progi1984/issue29293

BO : Fixed redirection when permission is not allowed

jolelievre committed:
Merge pull request #39321 from jolelievre/wall-of-fame-v8

Update Distribution API Client to include the new wall of fame

jolelievre committed:
Merge pull request #39322 from jolelievre/wall-of-fame-v9

Update Distribution API Client to include the new wall of fame

boherm committed:
Delete symlink for apiplatform bundle

boherm committed:
Add new service to install sf bundle assets

boherm committed:
Install bundles assets in finalize install step

jolelievre committed:
Add endpoints for attribute groups

jolelievre committed:
Optimize the product API UI tests, the API client is created previously and variables are set in environment

PrestaEdit committed:
fix: using dev version of ps_apiresources

PrestaEdit committed:
fix: update composer.lock

PrestaEdit committed:
Add endpoints for attribute groups

aomaxime committed:
Fix product redirection when delete

Commit:
Install : Configure : Hide password error on keyup

Codencode committed:
Correct tag closing in AboutCommand.php

Hlavtox committed:
Merge pull request #39272 from Progi1984/issue35205

BO : Fixed the memorization of the language used in Admin in the cookie (if option is enabled)

Hlavtox committed:
Merge pull request #39271 from Progi1984/issue37614

BO : TinyMCE : Fixed checkbox icon

Hlavtox committed:
Merge pull request #39269 from Progi1984/issue35124

BO : Admin API : Fixed selectors for copying secret

Codencode committed:
Added self::SUCCESS in AboutCommand.php"

Codencode committed:
Prevents form submission with Enter on all fields except textareas

jolelievre committed:
Merge pull request #39331 from PrestaEdit/patch-948339

Fix: using dev version of ps_apiresources

jolelievre committed:
Adapt client credentials flow tests to avoid removing the initial API client

jolelievre committed:
Update CQRS discount commands and queries to handle the carriers

jolelievre committed:
Optimize customer groups API ui tests

jolelievre committed:
Integrate form type for carriers condition

M0rgan01 committed:
fix-shipments-items

jolelievre committed:
Handle delivery carriers conditions in discoutn form data handler and provider

jolelievre committed:
Optimize hooks API ui tests

jolelievre committed:
Optimize languages API ui tests

jolelievre committed:
Optimize modules API ui tests

jolelievre committed:
Adapt and optimize API client UI tests

jolelievre committed:
Update nightly workflow

kpodemski committed:
Merge pull request #39299 from Progi1984/issue21262

tyloo committed:
chore: Fixed version in the version badge

Signed-off-by: Julien Bonvarlet <julien.bonvarlet@prestashop.com>

Hlavtox committed:
Merge pull request #39311 from Hlavtox/unify-error-handling-9.0.x

Backport Unify error handling in FO and get rid of Tools::displayError calls

Hlavtox committed:
Merge pull request #39196 from Hlavtox/dont-prompt-debug

Change 404 admin template

jolelievre committed:
Fix keycloak tests

jolelievre committed:
Update README to setup API env variables

jolelievre committed:
Add behat test for carrier trigger on FO

jolelievre committed:
Clean force configuration PS_CART_RULE_FEATURE_ACTIVE in behat tests since it is automatically updated by CartRule object model on creation

jolelievre committed:
Merge pull request #39329 from jolelievre/update-api-module

Update api module and optimise API UI tests

jolelievre committed:
Merge branch '9.0.x' into merge-90-develop-110825

Hlavtox committed:
Merge pull request #39337 from Codencode/fix-39319-Cannot-use-newline-in-textarea-on-product-page

Fix: Cannot use newline in textarea on product page

kpodemski committed:
Merge pull request #39245 from Codencode/fix-39244-Back-Office-becomes-inaccessible-if-the-employee's-assigned-language-is-deleted

Fix: Back Office becomes inaccessible if the employee's assigned language is deleted

boherm committed:
Merge pull request #39314 from Progi1984/issue26860

boherm committed:
Add admin-dev bundles folder in gitignore

jolelievre committed:
Merge pull request #39353 from jolelievre/merge-90-develop-110825

Merge 9.0.x to develop 11/08/25

jolelievre committed:
Merge pull request #39338 from jolelievre/discount-carriers

Update CQRS discount commands and queries to handle the carriers

jolelievre committed:
Merge pull request #39246 from tleon/Issue-38674-discount-condition-based-on-supplier

Discount condition based on supplier

Hlavtox committed:
Merge pull request #39277 from Codencode/idea-39259-Make-the-TinyMCEEditor-more-flexible-and-extensible-via-module-for-ps-9

Make the TinyMCEEditor more flexible and extensible via module

jolelievre committed:
Merge pull request #39341 from M0rgan01/fix-sipments-items

Fix shipments "items"

aomaxime committed:
fix rebase

tleon committed:
chore(discounts): condition based on countries

ShaiMagal committed:
Update Connection.php

ShaiMagal committed:
Update Connection.php

PSebou committed:
update with guest form

PSebou committed:
update with guest form

Touxten committed:
Merge pull request #39287 from Hlavtox/fix-last-positions

Fix + simplify last positions and add comments to it

Hlavtox committed:
Merge pull request #39333 from aomaxime/fix-product-delete-redirection-seo

Fix product access in BO when seo redirection related entity is deleted

alexandrebru83 committed:
add hook on rendertemplate in CheckoutStep

ShaiMagal committed:
Update Store.php

kpodemski committed:
Merge pull request #39364 from ShaiMagal/seznambot

kpodemski committed:
Merge pull request #39373 from ShaiMagal/store_refacto

kpodemski committed:
Merge pull request #39349 from tyloo/chore/fix-readme-version

chore: Fixed version in the README version badge

PSebou committed:
update validate firstname and lastname

PSebou committed:
update validate firstname and lastname

Touxten committed:
No delete default country

Touxten committed:
Csfixer

jolelievre committed:
Merge pull request #39327 from boherm/fix-symlink-for-apiplatform

Fix symlink issues on install

PSebou committed:
Merge branch 'refs/heads/feature/opc-optional-fields' into feature/Fulfill-the-contact-information

Touxten committed:
Merge pull request #39375 from Touxten/no-delete-default-country

Don't allow  delete default country

Touxten committed:
Merge pull request #39336 from Progi1984/issue31224

Install : Configure : Hide password error on keyup

Touxten committed:
Merge pull request #39340 from aomaxime/fix-category-delete-redirection-seo

 Fix product / category access in BO when seo redirection related category is deleted

Touxten committed:
Merge pull request #39366 from alexandrebrubeyonds/hook/actionCheckoutStepRenderTemplateRebase

add hook on rendertemplate in CheckoutStep

PSebou committed:
update form login and customer

PSebou committed:
update form login and customer

jolelievre committed:
Merge branch '8.2.x' into merge-82-90-190825

jolelievre committed:
Merge pull request #39385 from jolelievre/merge-82-90-190825

Merge 8.2.x to 9.0.x 19/08/25

jolelievre committed:
Merge branch '9.0.x' into merge-90-develop-190825

jolelievre committed:
Delay kernel creation in install init script to get full access to required constants

jolelievre committed:
Define fallback values in the console when no shop is installed and/or no parameters defined

jolelievre committed:
Handle product conditions based on attributes in CQRS commands/queries, refactor the behat context when multiple criterias are set they should be in separated ProductRuleGroup representing an AND condition that is more restrictive

kpodemski committed:
Merge pull request #39197 from Codencode/idea-39126-Add-PrestaShop-version-in-bin/console

Added PrestaShop version in bin/console

Hlavtox commented on classes/controller/FrontController.php:
```diff
@@ -1996,17 +2017,25 @@ protected function makeCustomerFormatter()
     {
-        $guestAllowedCheckout = Configuration::get('PS_GUEST_CHECKOUT_ENABLED');
+        if($opc){
+            $guestAllowedCheckout = false;
```
Why is guest checkout on OPC always false?

PSebou replied:
because we have 3 form for OPC

1. Guest form with just email
2. Login form 
3. new customer form. For this one we are sure it's not guest part because we purpose guest form first. So if click to create account you don't want to be guest.

Hlavtox reviewed (commented):
@PSebou @matks I think we should avoid using some specific name for the controller - `OnePageCheckoutController`.

Because if we decide, and I think it will lead to it, that the openpagecheckout will be the only one, we will end up with a complicated name.

So, what about just `CheckoutController`?

kpodemski commented on classes/controller/FrontController.php:
```diff
@@ -1967,7 +1968,7 @@ protected function getTranslator()
     }
 
-    protected function makeLoginForm()
+    protected function makeLoginForm($opc = false)
```
Wouldn't it be better to do

```
$loginForm = $this->makeLoginForm
$loginForm->setAction('YourCustomUrl');
```

?

PSebou replied:
hmm I don't know, I can change if you want.

jolelievre replied:
Agreed if you need to change this only for your POCController, then no need to change it in the parent controller

You can override the makeLoginForm where it's relevant. By the way why do you need to add these extra fields? Why does the OPC impact the way people login?

kpodemski commented on classes/controller/FrontController.php:
```diff
@@ -1977,12 +1978,32 @@ protected function makeLoginForm()
         );
 
-        $form->setAction($this->getCurrentURL());
+        $form->setAction($this->getCurrentURL().($opc?'?ajax=1&submitLogin=1':''));
```
code format + see a comment above

kpodemski commented on classes/controller/FrontController.php:
```diff
@@ -1977,12 +1978,32 @@ protected function makeLoginForm()
+        return $form;
+    }
+
+    protected function makeGuestForm($opc = false)
```
similar to the login form

jolelievre replied:
This can be moved only in OPCController

kpodemski commented on classes/controller/FrontController.php:
```diff
@@ -1977,12 +1978,32 @@ protected function makeLoginForm()
     }
 
-    protected function makeCustomerFormatter()
+    protected function makeCustomerFormatter($opc = false)
```
again, you have an instance of formatted in your controller, so you don't need to have this parameter imho

kpodemski commented on classes/form/CustomerForm.php:
```diff
@@ -269,6 +269,10 @@ public function submit()
 
     public function getTemplateVariables()
     {
+        if (!$this->formFields) {
```
that's a bit odd, what's the reason for this?

PSebou replied:
It's when fillWith is not execute. On CustomerLoginForm I found a similar code

jolelievre replied:
Why not call `fillWith` explicitely in your controller then? Instead of adding some magic automatic filling that can have some impacts in other use cases?

kpodemski commented on classes/form/GuestForm.php:
```diff
@@ -0,0 +1,215 @@
+/**
+ * StarterTheme TODO: B2B fields, Genders, CSRF.
+ */
+class GuestFormCore extends AbstractForm
```
If you edit only a part of what's inside CustomerFormCore, then why not extend it in the first place?

jolelievre replied:
Agreed, you need to avoid duplicating as much as possible

kpodemski commented on controllers/front/OnePageCheckoutController.php:
```diff
@@ -0,0 +1,540 @@
+use PrestaShop\PrestaShop\Core\Foundation\Templating\RenderableProxy;
+use PrestaShopBundle\Translation\TranslatorComponent;
+
+class OnePageCheckoutControllerCore extends FrontController
```
why not extend CheckoutController and adjust what's needed?

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+ */
+import $ from 'jquery';
+
+function isValidEmail(email) {
```
make sure to scope your functions, having global isValidEmail may not be safe for the sake of compatibility across ecosystem

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+  } else if($('#customer-form #field-password').val() === ''){
+    console.log('password is empty');
+    return false;
+  } else if($('#customer-form #field-firstname').val() === ''){
```
there's no reason to do `#customer-form #field-lastname`

`#field-lastname` must be unique thus targeting via parent selector doesn't make sense

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+
+function checkCustomerFormIsValid(){
+
+  if(!isValidEmail($('#customer-form #field-email').val())){
```
I highly suggest thinking about some lighweight library to do the work on the JS validation, if you want to have one

PSebou replied:
I don't want to have js validation. Just now when trigger the ajax call

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+      if (isValidEmail && email !== lastSentEmail) {
+        lastSentEmail = email;
+
+        prestashop.emit('GuestEmailEntered', {
```
maybe you could prefix the events somehow

opc.guestEmailEntered, etc.

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+    }, 5000);
+  });
+
+  $('#login-form input#field-email').on('input', function () {
```
id is unique, no need to target like this

it's just #field-email

that being said, the good practice is to rely on `js-` selectors to indicate that parts of the system are related to JS logic

PSebou replied:
No with the form generator all email field have the same ID

kpodemski commented on themes/_core/js/one-page-checkout.js:
```diff
@@ -0,0 +1,272 @@
+})
+
+prestashop.on('GuestEmailEntered', (event) => {
+  prestashop.email = event.detail.email;
```
you could think of making an

OnePageCheckout object that is available through `window`, I can imagine that some of the developers would like to extend some of the features behind OPC and not modify the core files directly

this is very important

kpodemski commented on themes/one-page-checkout.js:
```diff
@@ -0,0 +1,18 @@
+(()=>{var e={"./node_modules/jquery/dist/jquery.js":
```
compiled assets should not be part of the commit

kpodemski committed:
Export module translations command

refacto

fix

cleaning

fix

fix

fix

do not use translator

fix php-cs-fixer

fix phpstan

support multi-locale export

cs fixer

phpstan fix

kpodemski committed:
Merge pull request #39405 from kpodemski/export-module-translations

Export module translations command

jolelievre committed:
Create new generic type GroupedItemCollectionType, along with its TypeScript and Vue app that handle a selection modal and an input tags rendering

jolelievre committed:
Integrate attributes selection in the discount form using the new GroupedItemsType component

jolelievre committed:
Add new ProductRuleGroup::type field that allows handling different type of validations for product rules, the default one is the legacy one at_leat_one_product_rule And a new type was added all_product_rules that requires all product rules inside a group to be valid for the group to be considered valid

Krinkle committed:
Update CSSJanus to wikimedia/cssjanus

Ref https://packagist.org/packages/cssjanus/cssjanus
Ref https://packagist.org/packages/wikimedia/cssjanus

jolelievre committed:
Merge pull request #39363 from tleon/Issue-38671-trigger-based-on-country

Condition based on countries

jolelievre committed:
Refacto discount form after the GroupedItemCollection form type refacto

jolelievre committed:
Merge pull request #39388 from jolelievre/merge-90-develop-190825

Merge 9.0.x to develop 19/08/25

Touxten committed:
Merge pull request #39424 from Krinkle/patch-vendor

Update CSSJanus to wikimedia/cssjanus

kpodemski committed:
Merge pull request #39418 from jolelievre/discount-attributes-condition

Discount attributes condition in form

PSebou commented:
for this question : https://github.com/PrestaShop/PrestaShop/pull/39187#pullrequestreview-3137220585. At this moment we are pretty sure we will keep both : OPC and 5 steps. I agree with you if we will remove 5 steps it can be good to change onepagecheckout by checkout. Actually we will keep both.

jolelievre committed:
Fix docker initialisation

PSebou committed:
treat feedback and clean code format

PSebou committed:
treat feedback and clean code format

PSebou committed:
creation of checkoutController to have master Class for OrderController.php and OnePageCheckoutController.php wip

PSebou committed:
creation of checkoutController to have master Class for OrderController.php and OnePageCheckoutController.php wip

jolelievre commented on install-dev/data/xml/meta.xml:
```diff
@@ -120,6 +120,10 @@
 			<page>cms</page>
 			<configurable>0</configurable>
 		</meta>
+    <meta id="one-page-checkout">
```
Please fix the indentation here

jolelievre commented on classes/controller/CheckoutController.php:
```diff
@@ -0,0 +1,433 @@
+use PrestaShop\PrestaShop\Core\Foundation\Templating\RenderableProxy;
+use PrestaShopBundle\Translation\TranslatorComponent;
+
+class CheckoutControllerCore extends FrontController
```
Please add some extensive description, that this class is now the base class for both OnePageCheckout and regular/legacy multi-step checkout, and that the OrderController is the alternative controller

The use of both controllers depends on some configuration

jolelievre replied:
Also I think you copied all the code of OrderController here and left it empty What would be interesting would be to keep the specificities of the OrderControler inside it, basically all the methods you need to override in your OPCController should also be overridden in the OrderController this way we look at the two classes we can see exactly what are their differences

And then this new CheckoutController can be turned into an abstract class, and maybe even force some abstract methods that MUST be implemented by its daughter classes

PSebou replied:
Ok I take note about the checkoutController I will maybe implement it on an other PR

jolelievre commented on install-dev/data/db_structure.sql:
```diff
@@ -609,15 +609,15 @@ CREATE TABLE `PREFIX_customer` (
   `id_shop_group` INT(11) UNSIGNED NOT NULL DEFAULT '1',
   `id_shop` INT(11) UNSIGNED NOT NULL DEFAULT '1',
-  `id_gender` int(10) unsigned NOT NULL,
+  `id_gender` int(10) unsigned DEFAULT NULL,
```
I'm still very not at ease with these modifications In many legacy places there is very little checks and the code or the template simply espects the data to be present and filled

So it accesses it without care and this kind of modification could very easily create some side effects simply because it wants to access the first letter of of the first name for example

jolelievre replied:
I already said this in your previous PR but you should keep retro compatibility as your first goal for this sensitive modification of the OPC So, you shouldn't need to change the model constraints or the db tables simply because you change the checkout form

Instead you need to think about a solution to keep the data consistent with the multi steps checkout meaning:
- you must define a gender, even if you select one by default You could even create a new "Non-gender" title in the default fixtures
- you must fill firstname and lastname, either with placeholders if you have no othger solution but when I look at the screenshot I'm almost 100% sure that you can get these two fields in the second part of the form when the address is filled

PSebou replied:
I agree, It's not a good idea to change the database structure. For Firstname and lastname, maybe we can implement it later

jolelievre reviewed (changes requested):
Thanks @PSebou 

it's looking better and better, there is still some stuff to improve, but you're on better tracks

PSebou committed:
Merge branch 'feature/Fulfill-the-contact-information' of github.com:/PSebou/PrestaShop into feature/Fulfill-the-contact-information

PSebou committed:
fix phpcs

PSebou committed:
fix feedbacks

PSebou commented:
Sorry I have an issue with rebase I change the branch now the new PR is https://github.com/PrestaShop/PrestaShop/pull/39530
